### PR TITLE
Bug 1974830: Update KubeDeploymentReplicasMismatch alert

### PR DESCRIPTION
**Pull request by PhilipGough**
* [ ] I added CHANGELOG entry for this change.
* [x] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by paulfantom** [comment]
/lgtm
/hold

Putting on hold to more allow testing in CI.

**Review by jan--f — APPROVED** [review]
lgtm :+1:

File: jsonnet/rules.libsonnet [hunk 1]
```diff
@@ -343,15 +350,15 @@ local droppedKsmLabels = 'endpoint, instance, job, pod, service';
...
                 kube_deployment_spec_replicas{namespace=~"(openshift-.*|kube-.*|default|logging)",job="kube-state-metrics"}
-                  !=
+                  >
                 kube_deployment_status_replicas_available{namespace=~"(openshift-.*|kube-.*|default|logging)",job="kube-state-metrics"}
               ) and (
                 changes(kube_deployment_status_replicas_updated{namespace=~"(openshift-.*|kube-.*|default|logging)",job="kube-state-metrics"}[5m])
                   ==
                 0
-              ) and cluster:control_plane:all_nodes_ready
+              )) * scalar(cluster:control_plane:all_nodes_ready)) != 0
```
Comment (simonpasquier) [inline jsonnet/rules.libsonnet:361]: (nit) I found it easier to parse in this form:

```suggestion
              )) * scalar(cluster:control_plane:all_nodes_ready == 1)
```

File: jsonnet/rules.libsonnet [hunk 2]
```diff
@@ -343,15 +350,15 @@ local droppedKsmLabels = 'endpoint, instance, job, pod, service';
           },
           {
             expr: |||
-              (
+              (((
                 kube_deployment_spec_replicas{namespace=~"(openshift-.*|kube-.*|default|logging)",job="kube-state-metrics"}
-                  !=
+                  >
```
Comment (paulfantom) [inline jsonnet/rules.libsonnet:355]: note: let's make sure this goes upstream :)

**Comment by simonpasquier** [comment]
/lgtm

After discussing offline with @pgough we understand why the former PR #1245 didn't work.  `cluster:control_plane:all_nodes_ready` was returning 1 only when all control plane nodes were ready, otherwise it wouldn't return any data (e.g. Prometheus would insert stale markers). And calling `scalar()` on an instant vector which doesn't have exactly one element (e.g. empty) returns `NaN`. Thus the full expression `... * scalar(cluster:control_plane:all_nodes_ready)) != 0` returned `NaN` too which is enough to trigger the alert.

**Pull request merged**